app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -20,28 +22,73 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
     private val refresh = object : Runnable {
         override fun run() {
             SensorWorker.start(requireContext())
+            totalDisabledSensors = 0
+            totalEnabledSensors = 0
             val sensorDao = AppDatabase.getInstance(requireContext()).sensorDao()
-            SensorReceiver.MANAGERS.forEach { managers ->
-                managers.availableSensors.forEach { basicSensor ->
+            SensorReceiver.MANAGERS.sortedBy { it.name }.filter { it.hasSensor(requireContext()) }.forEach { managers ->
+                managers.availableSensors.sortedBy { it.name }.forEach { basicSensor ->
                     findPreference<Preference>(basicSensor.id)?.let {
                         val sensorEntity = sensorDao.get(basicSensor.id)
+                        val sensorSettings = sensorDao.getSettings(basicSensor.id)
+                        if (!sensorSettings.isNullOrEmpty()) {
+                            sensorSettings.forEach { setting ->
```

Comment:
It would probably be better to add a DAO query to just get all the sensor settings, or better still one that just returns the list of names of settings since that's all you actually need rather than loop through and request individually.  That way we make one query when the screen loads rather than one query per sensor per update  But I'm still on the fence with how useful it is to show off all the sensors with settings.

Reply:
Yea I am not certain about it either, I can remove it for now so we only focus on enabling and disabling them all.  Maybe as the settings grow more we can decide a better way to manage them as a whole.